File: myportal/src/main/java/swp/portal/AgpWarenkorb.java
package swp.portal;

import com.vaadin.flow.templatemodel.TemplateModel;
import com.vaadin.flow.component.Tag;
import com.vaadin.flow.component.button.Button;
import com.vaadin.flow.component.dependency.JsModule;
import com.vaadin.flow.component.grid.Grid;
import com.vaadin.flow.component.html.Label;
import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
import com.vaadin.flow.component.orderedlayout.VerticalLayout;
import com.vaadin.flow.component.polymertemplate.Id;
import com.vaadin.flow.component.polymertemplate.PolymerTemplate;
import com.vaadin.flow.router.Route;

/**
 * A Designer generated component for the agp-warenkorb template.
 *
 * Designer will add and remove fields with @Id mappings but
 * does not overwrite or otherwise change this file.
 */
@Tag("agp-warenkorb")
@JsModule("./src/agp-warenkorb.js")
@Route("warenkorb")
public class AgpWarenkorb extends PolymerTemplate<AgpWarenkorb.AgpWarenkorbModel> {

	@Id("agpMenu")
	private AgpMenu agpMenu;
	@Id("vaadinVerticalLayout")
	private VerticalLayout vaadinVerticalLayout;
	@Id("vaadinGrid")
	private Grid vaadinGrid;
	@Id("buttonKaufen")
	private Button buttonKaufen;
	@Id("labelGesamtPreis")
	private Label labelGesamtPreis;
	
    public AgpWarenkorb() {
        // You can initialise any data required for the connected UI components here.
    	
//        vaadinGrid.addColumn(WarenkorbTO??::getName).setHeader("Artikel");
//        vaadinGrid.addColumn(??::getAnzahl).setHeader("Anzahl");
//        vaadinGrid.addColumn(??::getEinzelpreis).setHeader("Einzelpreis");
//        vaadinGrid.addColumn(??::getGesamtpreis).setHeader("Gesamtpreis");
//        vaadinGrid.addColumn(??::getLoeschen).setHeader("Loeschen");
    	
    }
    
    

    /**
     * This model binds properties between AgpWarenkorb and agp-warenkorb
     */
    public interface AgpWarenkorbModel extends TemplateModel {
        // Add setters and getters for template properties here.
    }
}

File: myportal/frontend/src/agp-warenkorb.js
import {html, PolymerElement} from '@polymer/polymer/polymer-element.js';
import '@vaadin/vaadin-ordered-layout/src/vaadin-vertical-layout.js';
import './agp-menu.js';
import '@vaadin/vaadin-grid/src/vaadin-grid.js';
import '@vaadin/vaadin-button/src/vaadin-button.js';

class AgpWarenkorb extends PolymerElement {

    static get template() {
        return html`
<style include="shared-styles">
                :host {
                    display: block;
                    height: 100%;
                }
            </style>
<vaadin-vertical-layout style="width: 100%; height: 100%;" id="vaadinVerticalLayout">
 <agp-menu id="agpMenu"></agp-menu>
 <vaadin-grid id="vaadinGrid"></vaadin-grid>
 <label id="labelGesamtPreis">Gesamtpreis:</label>
 <vaadin-button theme="tertiary" id="buttonKaufen">
  Jetzt kaufen
 </vaadin-button>
</vaadin-vertical-layout>
`;
    }

    static get is() {
        return 'agp-warenkorb';
    }

    static get properties() {
        return {
            // Declare your properties here.
        };
    }
}

customElements.define(AgpWarenkorb.is, AgpWarenkorb);
